synapse/axon.py

```diff
@@ -875,8 +888,27 @@ async def get(self, sha256):
         fhash = s_common.ehex(sha256)
         logger.debug(f'Getting blob [{fhash}].', extra=await self.getLogExtra(sha256=fhash))
 
-        async for byts in self._get(sha256):
-            yield byts
+        if offs is not None or size is not None:
+
+            if not self.byterange:  # pragma: no cover
+                mesg = 'This axon does not support byte ranges.'
+                raise s_exc.FeatureNotSupported(mesg=mesg)
+
+            if offs < 0:
+                raise s_exc.BadArg(mesg='Offs must be >= 0', offs=offs)
+            if size < 1:
+                raise s_exc.BadArg(mesg='Size must be >= 1', size=size)
+
+            fsize = await self.size(sha256)
+            if offs > fsize:
```

Comment:
In most file-like APIs this should just read `b''` rather than returning an error. maybe we should just return?